src/window-event-handler.coffee

```diff
@@ -24,13 +23,12 @@ class WindowEventHandler
             if pathToOpen? and needsProjectPaths
               if fs.existsSync(pathToOpen)
                 atom.project.addPath(pathToOpen)
-              else if fs.existsSync(path.dirname(pathToOpen))
-                atom.project.addPath(path.dirname(pathToOpen))
               else
-                atom.project.addPath(pathToOpen)
+                dirToOpen = path.dirname(pathToOpen)
+                if fs.existsSync(dirToOpen)
+                  atom.project.addPath(dirToOpen)
 
-            {protocol} = url.parse(pathToOpen)
-            unless fs.isDirectorySync(pathToOpen) or protocol?
+            unless fs.isDirectorySync(pathToOpen)
```

Comment:
I think that we always need to open it. Other packages could define other 'opener` URIs, like `markdown-preview://view`, or whatever. In general, remote URIs could represent files, not just directories.